mongo/integration/main.go

```diff
@@ -1,7 +0,0 @@
-// Copyright (C) MongoDB, Inc. 2017-present.
-//
-// Licensed under the Apache License, Version 2.0 (the "License"); you may
-// not use this file except in compliance with the License. You may obtain
-// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
-
-package integration
```

Comment:
The compile tests for the CI are failing for 

```
go build go.mongodb.org/mongo-driver/mongo/integration: build constraints exclude all Go files in /data/mci/{redacted}/src/go.mongodb.org/mongo-driver/mongo/integration
```

It looks like the only files in  `mongo/integration` are `*_test.go` files which [are ignored on build](https://pkg.go.dev/cmd/go#hdr-Test_packages:~:text=When%20compiling%20packages%2C%20build%20ignores%20files%20that%20end%20in%20%27_test.go%27.). These main.go files seem to act as a kind of "shim" file so that the packages compile with `make build`. Perhaps the right course of actions is to exclude them from the packages returned by the `etc/list_pkgs.sh` script. Though, one downside of this is that it will be the source of a (probably) frustrating bug if we ever add non test code to this package. What are your thoughts?

Reply:
Thanks for getting to the bottom of the build failures! It seems that listing the packages to use that as an input to the `go build` command (and other commands) isn't necessary and could instead be replaced by a package glob expression like `go.mongodb.org/mongo-driver/...`. I'm going to investigate updating the Makefile (and some other scripts) to use package glob expressions instead of a list of packages.

Reply:
For some reason, the build using package glob expression works on newer Go versions, but fails on Go 1.10. I'm going to just re-add the `main.go` file and add a comment describing why it exists. I may try some of the build script improvements in a subsequent PR, but this doesn't seem like the right place for more major build script updates.